src/cocotb/__init__.py

```diff
@@ -432,3 +357,48 @@ def _stop_user_coverage() -> None:
         _user_coverage.stop()
         cocotb.log.debug("Writing coverage data")
         _user_coverage.save()
+
+
+def _setup_random_seed() -> None:
+    global _random_seed
+
+    seed_envvar = os.getenv("RANDOM_SEED")
+    if seed_envvar is None:
+        if "ntb_random_seed" in plusargs:
+            plusarg_seed = plusargs["ntb_random_seed"]
+            if not isinstance(plusarg_seed, str):
+                raise TypeError("ntb_random_seed plusarg is not a valid seed value.")
+            _random_seed = int(plusarg_seed)
+        elif "seed" in plusargs:
+            plusarg_seed = plusargs["seed"]
+            if not isinstance(plusarg_seed, str):
+                raise TypeError("ntb_random_seed plusarg is not a valid seed value.")
+            _random_seed = int(plusarg_seed)
+        else:
+            _random_seed = int(time.time())
+        log.info("Seeding Python random module with %d" % (_random_seed))
+    else:
+        _random_seed = int(seed_envvar)
+        log.info("Seeding Python random module with supplied seed %d" % (_random_seed))
+
+    random.seed(_random_seed)
+
+
+def _setup_root_handle() -> None:
+    root_name = os.getenv("TOPLEVEL")
+    if root_name is not None:
+        root_name = root_name.strip()
+        if root_name == "":
+            root_name = None
+        elif "." in root_name:
+            # Skip any library component of the toplevel
+            root_name = root_name.split(".", 1)[1]
+
+    from cocotb import simulator
+
+    handle = simulator.get_root_handle(root_name)
+    if not handle:
+        raise RuntimeError(f"Can not find root handle ({root_name})")
```

Comment:
```suggestion
        raise RuntimeError(f"Can not find root handle {root_name!r}")
```
Alternative to the suggestion by @marlonjames, we have that syntax elsewhere.